src/main/java/uk/gov/hmcts/divorce/caseworker/service/ReIssueApplicationService.java

```diff
@@ -0,0 +1,126 @@
+        ReissueOption reissueOption = caseDetails.getData().getApplication().getReissueOption();
+
+        log.info("For case id {} reissue option selected is {} ", caseDetails.getId(), reissueOption.getLabel());
+
+        caseDetails.getData().getApplication().setReissueDate(now(clock));
+
+        CaseDetails<CaseData, State> updatedCaseDetails = null;
+
+        switch (reissueOption) {
+            case DIGITAL_AOS:
+                log.info("For case id {} processing reissue for digital aos ", caseDetails.getId());
+                updatedCaseDetails = caseTasks(
+                    generateRespondentSolicitorAosInvitation,
+                    generateCitizenRespondentAosInvitation,
+                    sendAosNotifications,
+                    details -> {
+                        details.getData().getApplication().setReissueDate(now(clock));
+                        details.getData().setDueDate(now(clock).plusDays(daysUntilOverdue));
+                        return details;
+                    },
+                    setPostIssueState,
+                    sendApplicationIssueNotifications
+                ).run(caseDetails);
+                break;
+
+            case OFFLINE_AOS:
+                log.info("For case id {} processing reissue for offline aos ", caseDetails.getId());
+                updatedCaseDetails = caseTasks(
+                    generateRespondentSolicitorAosInvitation,
+                    generateCitizenRespondentAosInvitation,
+                    sendAosPack,
+                    details -> {
+                        details.getData().getApplication().setReissueDate(now(clock));
+                        details.getData().setDueDate(now(clock).plusDays(daysUntilOverdue));
+                        return details;
+                    },
+                    setPostIssueState,
+                    sendApplicationIssueNotifications
+                ).run(caseDetails);
+                break;
+
+            case REISSUE_CASE:
+                log.info("For case id {} processing complete reissue ", caseDetails.getId());
+                updatedCaseDetails = caseTasks(
+                    generateMiniApplication,
+                    generateRespondentSolicitorAosInvitation,
+                    generateCitizenRespondentAosInvitation,
+                    sendAosNotifications,
+                    sendAosPack,
+                    details -> {
+                        details.getData().getApplication().setReissueDate(now(clock));
+                        details.getData().setDueDate(now(clock).plusDays(daysUntilOverdue));
+                        return details;
+                    },
+                    setPostIssueState,
+                    sendApplicationIssueNotifications
+                ).run(caseDetails);
+                break;
+
+            default:
+                log.info("For case id {} invalid reissue option hence not processing reissue application ", caseDetails.getId());
```

Comment:
I know this should never happen but you will get a null pointer error here because updatedCaseDetails is null and you use it below.